### ISSUE TITLE
warning if configure file and work dir  on different windows partitions 

### ISSUE THREAD

**Issue by wu0407** [golangci/golangci-lint#2368]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Run golangci-lint on code dir (D:\git\k8s-gateway-service\internal\server\cluster) but specified config in  vscode user cofigure dir(C:\Users\xiaoqing\AppData\Roaming\Code\User\golangci.yml) . 
it report not use any configure file `level=info msg="[config_reader] Used config file "`.

It return errors:
```
PS D:\git\k8s-gateway-service\internal\server\cluster> d:\go\bin\golangci-lint.exe run --enable revive --config $env:APPDATA/Code/User/golangci.yml  -v
level=warning msg="[config_reader] Can't pretty print config file path: can't get relative path for path C:\\Users\\xiaoqing\\AppData\\Roaming\\Code\\User\\golangci.yml and root D:\\git\\k8s-gateway-service\\internal\\server\\cluster: Rel: can't make C:\\Users\\xiaoqing\\AppData\\Roaming\\Code\\User\\golangci.yml relative to D:\\git\\k8s-gateway-service\\internal\\server\\cluster"
level=info msg="[config_reader] Used config file "
level=info msg="[lintersdb] Active 11 linters: [deadcode errcheck gosimple govet ineffassign revive staticcheck structcheck typecheck unused varcheck]"
level=info msg="[loader] Go packages loading at mode 575 (compiled_files|deps|files|name|exports_file|imports|types_sizes) took 709.842ms"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 1.0011ms"
level=info msg="[linters context/goanalysis] analyzers took 71.9811ms with top 10 stages: buildir: 13ms, SA1012: 999.7µs, SA6005: 999.7µs, errorsas: 999.7µs, SA5003: 999.7µs, SA5002: 999.7µs, SA2001: 999.7µs, S1031: 999.7µs, S1020: 999.7µs, S1012: 999.7µs"
level=warning msg="[runner] Can't run linter goanalysis_metalinter: the_only_name: cannot find rule: time-equal"
level=info msg="[runner] processing took 0s with stages: exclude-rules: 0s, uniq_by_line: 0s, path_shortener: 0s, skip_dirs: 0s, autogenerated_exclude: 0s, exclude: 0s, diff: 0s, max_per_file_from_linter: 0s, severity-rules: 0s, sort_results: 0s, path_prettifier: 0s, source_code: 0s, path_prefixer: 0s, identifier_marker: 0s, filename_unadjuster: 0s, skip_files: 0s, nolint: 0s, max_same_issues: 0s, max_from_linter: 0s, cgo: 0s"
level=info msg="[runner] linters took 677.7169ms with stages: goanalysis_metalinter: 677.7169ms"
level=info msg="File cache stats: 0 entries of total size 0B"
level=info msg="Memory: 16 samples, avg is 86.5MB, max is 312.8MB"
level=info msg="Execution took 1.4383687s"
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.43.0 built from (unknown, mod sum: "h1:SLwZFEmDgopqZpfP495zCtV9REUf551JJlJ51Ql7NZA=") on (unknown)
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
linters-settings:
  revive:
    # see https://github.com/mgechev/revive#available-rules for details.
    ignore-generated-header: true
    severity: warning
    rules:
      - name: indent-error-flow
        severity: warning
      - name: add-constant
        severity: warning
        arguments:
          - maxLitCount: "3"
            allowStrs: '""'
            allowInts: "0,1,2"
            allowFloats: "0.0,0.,1.0,1.,2.0,2."
      - name: context-keys-type
      - name: time-equal
      - name: time-naming
      - name: var-declaration
      - name: unexported-return
      - name: errorf
      - name: blank-imports
      - name: context-as-argument
      - name: dot-imports
      - name: error-return
      - name: error-strings
      - name: error-naming
      - name: exported
      - name: increment-decrement
      - name: var-naming
      - name: package-comments
      - name: range
      - name: receiver-naming
      - name: empty-block
      - name: modifies-value-receiver
      - name: range-val-address
      - name: unhandled-error
      - name: string-of-int

```

</details>

### Go environment

<details>

```console
$ go version && go env
 go version go1.17.3 windows/amd64
go env
PS D:\git\k8s-gateway-service\internal\server\cluster> go env
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\xiaoqing\AppData\Local\go-build
set GOENV=C:\Users\xiaoqing\AppData\Roaming\go\env
set GOEXE=.exe
set GOEXPERIMENT=
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=D:\go\pkg\mod
set GONOPROXY=*.aaa.com
set GONOSUMDB=*.aaa.com
set GOOS=windows
set GOPATH=D:\go
set GOPRIVATE=*.aaa.com
set GOPROXY=https://goproxy.cn,direct
set GOROOT=D:\go-program
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=D:\go-program\pkg\tool\windows_amd64
set GOVCS=
set GOVERSION=go1.17.3
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=D:\git\k8s-gateway-service\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=C:\Users\xiaoqing\AppData\Local\Temp\go-build2478620999=/tmp/go-build -gno-record-gcc-switches
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
level=info msg="[config_reader] Config search paths: [./ D:\\git\\victoria-metrics-k8s-stack D:\\git D:\\ C:\\Users\\xiaoqing]"
level=info msg="[lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck]"
level=info msg="[loader] Go packages loading at mode 575 (compiled_files|types_sizes|deps|exports_file|files|imports|name) took 106.4468ms"
level=error msg="Running error: context loading failed: no go files to analyze"
level=info msg="Memory: 3 samples, avg is 43.9MB, max is 43.9MB"
level=info msg="Execution took 154.85ms"

```

</details>

### Code example or link to a public repository

<details>

```go
// add your code here
```

https://github.com/golang/vscode-go/issues/1901
</details>

**Comment by wu0407** [comment]
sorry ,it is only warnning.
